### PR TITLE
Memory-safety proofs for aws_priority_queue functions

### PR DESCRIPTION
**Pull request by feliperodri**
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

*Description of changes:*

Adds invariant for `aws_priority_queue` structure and proof harnesses for the following functions:
 - aws_priority_queue_init_dynamic
 - aws_priority_queue_init_static
 - aws_priority_queue_clean_up
 - aws_priority_queue_top
 - aws_priority_queue_size
 - aws_priority_queue_capacity

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .cbmc-batch/include/proof_helpers/utils.h [hunk 1]
```diff
@@ -57,3 +57,8 @@ void save_byte_from_hash_table(const struct aws_hash_table *map, struct store_by
  * Checks that a no bytes in the hash_table have changed from when "storage" was stored.
  */
 void check_hash_table_unchanged(const struct aws_hash_table *map, const struct store_byte_from_buffer *storage);
+
+/**
+ * Standard stub function to compare two items
+ */
+int s_compare(const void *a, const void *b);
```
Comment (ColdenCullen) [inline .cbmc-batch/include/proof_helpers/utils.h]: Can you rename this please? a) it's not actually static, so `s_` isn't appropriate and b) maybe make it clear that it's non-deterministic

Comment (feliperodri) [reply .cbmc-batch/include/proof_helpers/utils.h]: Thanks! Done!

File: source/priority_queue.c [hunk 2]
```diff
@@ -157,6 +157,16 @@ void aws_priority_queue_init_static(
     aws_array_list_init_static(&queue->container, heap, item_count, item_size);
 }
 
+bool aws_priority_queue_is_valid(struct aws_priority_queue *queue) {
```
Comment (karkhaz) [inline source/priority_queue.c]: `struct aws_priority_queue const * queue` please, or even `struct aws_priority_queue const * const queue` if possible.

File: .cbmc-batch/source/make_common_data_structures.c [hunk 3]
```diff
@@ -57,6 +57,23 @@ void ensure_array_list_has_allocated_data_member(struct aws_array_list *list) {
     }
 }
 
+bool aws_priority_queue_is_bounded(
+    struct aws_priority_queue *queue,
```
Comment (karkhaz) [inline .cbmc-batch/source/make_common_data_structures.c]: `aws_priority_queue const *queue` please, or even `aws_priority_queue const * const queue` if possible.

**Comment by feliperodri** [comment]
@karkhaz I addressed all your comments, thanks!

**Pull request merged**